### ISSUE TITLE
New Enhancement

### ISSUE THREAD

**Issue by MJZClwg** [victornguyen75/honey-encryption#3]
Hi i just want to ask if you have an updated code of honey encryption? and can you help me out on how can i separate the key from the shuffle passwords? Thank you :)

**Comment by victornguyen75** [comment]
Hi, unfortunately, I do not have an updated version for this code.

I'm not sure what you mean by separating the key from the shuffle passwords.

Are you talking about this code on line 129, and how to separate it?

`    passwords = list(passwordsToSeeds.keys())`

**Comment by MJZClwg** [comment]
Hi thankyou for your response. I already solved the one that i am asking.  I read some journals about honey encryption and it said that honey encryption has a message space.Can you help me out or give me some code on how can i apply that message space on your code? thankyou.

**Comment by victornguyen75** [comment]
Yeah sure I can help! 

_As a quick disclaimer, I'm not an amazing programmer, so I probably made some mistakes in my source code. If there is anything unclear, please feel free to ask me anything, and I'll do my best to clarify it._

Regarding your question about seed space, I do have a seed space in this code called `passwordsToSeeds`. I'm honestly not sure if this is right or not, but at the time of writing the code, having this dictionary `passwordsToSeeds` just made sense to me since I had a limited programming ability.

I put a reference at the end of this comment, so you know exactly where I'm getting my information. That way we can be on the same page of the facts.

Based on the journal below, it states the following:

> According to the probabilities of a message in the message space, it maps the message to a seed range in a seed space, then it randomly selects a seed from the range and XORs it with the key to get the ciphertext. 

**tl;dr, my code already includes the seed space and seed range, it's just a dictionary called `passwordsToSeeds`**. Now whether, using a dictionary is a good idea or not, I honestly don't know haha!

Regardless, does that help? Please feel free to let me know.

Reference: https://www.hindawi.com/journals/scn/2017/6760532/

**Comment by MJZClwg** [comment]
Hi,thanks for the answer it means a lot :) what about the message space because in journal that i read it says that you need to calculate the probability distribution form and the cumulative distribution form of a message. And do you have any suggestion or recommendation if we want to extend the message space from 10^8 to more larger space? thankyou in advance :)    On Thursday, December 6, 2018, 3:08:04 AM GMT+8, Victor Nguyen <notifications@github.com> wrote:  
 
 
Yeah sure I can help!

As a quick disclaimer, I'm not an amazing programmer, so I probably made some mistakes in my source code. If there is anything unclear, please feel free to ask me anything, and I'll do my best to clarify it.

Regarding your question about seed space, I do have a seed space in this code called passwordsToSeeds. I'm honestly not sure if this is right or not, but at the time of writing the code, having this dictionary passwordsToSeeds just made sense to me since I had a limited programming ability.

I put a reference at the end of this comment, so you know exactly where I'm getting my information. That way we can be on the same page of the facts.

Based on the journal below, it states the following:


According to the probabilities of a message in the message space, it maps the message to a seed range in a seed space, then it randomly selects a seed from the range and XORs it with the key to get the ciphertext.


tl;dr, my code already includes the seed space and seed range, it's just a dictionary called passwordsToSeeds. Now whether, using a dictionary is a good idea or not, I honestly don't know haha!

Regardless, does that help? Please feel free to let me know.

Reference: https://www.hindawi.com/journals/scn/2017/6760532/

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or mute the thread.
   

|  | Virus-free. www.avast.com  |

**Comment by MJZClwg** [comment]
Hi,thanks for the answer it means a lot :) what about the message space because in journal that i read it says that you need to calculate the probability distribution form and the cumulative distribution form of a message. And do you have any suggestion or recommendation if we want to extend the message space from 10^8 to more larger space? thankyou in advance :)

**Comment by MJZClwg** [comment]
and also what is the maximun message space of honey encryption?do have an idea?

**Comment by victornguyen75** [comment]
Mmmm. Yeah I didn't do any calculations for the probability distribution form and the cumulative distribution form of a message. I just created my message space without considering any calculations.

Regarding your questions about extending the message space, I don't have an idea about the maximum size or the recommended size. I'm not sure how much we will need.

If you can send me your calculations and references, I can gladly look over them and check your work.

Additionally, if you are able to incorporate them into the code, I can review your pull request.
